### PR TITLE
feat: Query to list Licence & Application numbers

### PR DESCRIPTION
**Pull request by hobbyhacker0**
## Description

This query is to provide Licence & Application ids (by org id) for use in create Conversation interface

Related issue: [VOL-4681](https://dvsa.atlassian.net/browse/VOL-4681)
Related issue: [VOL-4737](https://dvsa.atlassian.net/browse/VOL-4737)

## Before submitting (or marking as "ready for review")

- [x] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [x] Have you performed a self-review of the code
- [ ] Have you have added tests that prove the fix or feature is effective and working
- [ ] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

File: module/Api/src/Domain/QueryHandler/Messaging/ApplicationLicenceList/ByOrganisation.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+        $licences = $licenceRepository->fetchByOrganisationId($query->getOrganisation());
+
+        $applications = $applicationRepository->fetchByOrganisationIdAndStatuses(
+            $query->getOrganisation(),
+            Entity::ALL_APPLICATION_STATUSES,
+            AbstractQuery::HYDRATE_ARRAY
+        );
+
+        $results = array_fill_keys(['licences', 'applications'], array());
```
Comment (wadedvsa) [inline module/Api/src/Domain/QueryHandler/Messaging/ApplicationLicenceList/ByOrganisation.php:39]: S: `[]` over `array()`

File: module/Api/src/Domain/QueryHandler/Messaging/ApplicationLicenceList/ByApplicationToOrganisation.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            'organisation' => $organisation->getId()
+        ];
+
+        return $this->getQueryHandler()->handleQuery(GetApplicationListByOrganisationQuery::create($organisationQuery));
+    }
+
+    private function getApplicationRepository(): ApplicationRepository
+    {
+        $applicationRepository = $this->getRepo('Application');
+        assert($applicationRepository instanceof ApplicationRepository);
```
Comment (wadedvsa) [inline module/Api/src/Domain/QueryHandler/Messaging/ApplicationLicenceList/ByApplicationToOrganisation.php:43]: Ongoing discussion here but I think this one doesn't add any value as the return type is enforced.

**Review by wadedvsa — APPROVED** [review]
No required changes, just comments.

File: module/Api/src/Domain/QueryHandler/Messaging/ApplicationLicenceList/ByApplicationToOrganisation.php (hunk 2 above)
Comment (hobbyhacker0) [reply module/Api/src/Domain/QueryHandler/Messaging/ApplicationLicenceList/ByApplicationToOrganisation.php:43]: True in this case there's even less of an argument for it

**Pull request merged**